### PR TITLE
[auth-swift] TODO review and ObjC Sample delete

### PR DESCRIPTION
**Pull request by paulb777**
Review TODO's in the Auth Swift implementation of the library (not tests)

- Address some of them
- Delete already resolved ones
- Mark existing TODO's from the ObjC implementation with ObjC

Delete the Objective C sample since it depends on no-longer visible internal methods. The ObjC sample from the `main` branch can be used a reference for future Swift Sample app development.

### CODE REVIEW

File: FirebaseAuth/Sources/Swift/Auth/Auth.swift [hunk 1]
```diff
@@ -93,7 +93,7 @@ extension Auth: AuthInterop {
           strongSelf.autoRefreshTokens = true
           strongSelf.scheduleAutoTokenRefresh()
 
-          #if os(iOS) || os(tvOS) // TODO: Is a similar mechanism needed on macOS?
+          #if os(iOS) || os(tvOS) // TODO(from ObjC): Is a similar mechanism needed on macOS?
```
Comment (andrewheard) [inline FirebaseAuth/Sources/Swift/Auth/Auth.swift:96]: nit: `TODO(ObjC)` to match the ones below

File: FirebaseAuth/Sources/Swift/Utilities/AuthErrorUtils.swift [hunk 2]
```diff
@@ -551,7 +550,7 @@ class AuthErrorUtils: NSObject {
   #endif // os(iOS)
 
   static func recaptchaSDKNotLinkedError() -> Error {
-    // TODO(chuanr): point the link to GCIP doc once available.
+    // TODO(chuanr, ObjC): point the link to GCIP doc once available.
```
Comment (andrewheard) [inline FirebaseAuth/Sources/Swift/Utilities/AuthErrorUtils.swift:553]: nit: Is `ObjC` unique enough to search for these without `TODO`? If not we'll need to remember to search for `TODO(chuanr, ObjC)` and `TODO(ObjC)` to find these. (ignore this comment if it won't impact your workflow)

**Review by andrewheard — APPROVED** [review]

File: FirebaseAuth/Sources/Swift/Auth/Auth.swift (hunk 1 above)
Comment (paulb777) [reply FirebaseAuth/Sources/Swift/Auth/Auth.swift:96]: fixed

File: FirebaseAuth/Sources/Swift/Utilities/AuthErrorUtils.swift (hunk 2 above)
Comment (paulb777) [reply FirebaseAuth/Sources/Swift/Utilities/AuthErrorUtils.swift:553]: I would do `git grep TODO | grep ObjC`

**Pull request merged**